File: CustomerWebApi/Entities/Models/Customer.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace CustomerWebApi.Entities.Models
{
        [Table("Customer")]
        public class Customer
        {
            [Column("CustomerId")]
            public int Id { get; set; }

            [Required(ErrorMessage = "First Name is required")]
            [StringLength(60, ErrorMessage = "First Name can't be longer than 60 characters")]
            public string? FirstName { get; set; }
            [Required(ErrorMessage = "Last Name is required")]
            [StringLength(60, ErrorMessage = "Last Name can't be longer than 60 characters")]
            public string? LastName { get; set; }

            [Required(ErrorMessage = "Email is required")]
            public string? Email { get; set; }

            public DateTime Created { get; set; }

            public DateTime? Updated { get; set; }
        }
    
}

File: UI/CustomerClient/src/app/interfaces/Customer.ts
export interface Customer{
    id: number;
    firstName: string;
    lastName: string;
    email: string;
    created: string;
    updated: string;
    isEdit: boolean;
}


export const CustomerColumns = [  
    {
      key: 'id',
      type: 'number',
      label: 'Id',
      readonly: true
    },
    {
      key: 'firstName',
      type: 'text',
      label: 'First Name',
      required: true,
    },
    {
      key: 'lastName',
      type: 'text',
      label: 'Last Name',
      required: true,
    },
    {
      key: 'email',
      type: 'email',
      label: 'Email',
      required: true,
      pattern: '.+@.+',
    },
    {
      key: 'created',
      type: 'date',
      label: 'Created',
      readonly: true
    },
    {
        key: 'updated',
        type: 'date',
        label: 'Updated',
        readonly: true
      },
      {
        key: 'isEdit',
        type: 'isEdit',
        label: 'Action',
      },

  ];